pkg/profile/artifact.go

```diff
@@ -90,26 +80,51 @@ func (pa *ProfilesArtifactsMaker) Make(installation profilesv1.ProfileInstallati
 			}
 		}
 		for _, obj := range artifact.Objects {
-			filename := filepath.Join(artifactDir, fmt.Sprintf("%s.%s", obj.GetObjectKind().GroupVersionKind().Kind, "yaml"))
-			if err := pa.generateOutput(filename, obj); err != nil {
+			name := obj.GetObjectKind().GroupVersionKind().Kind
+			if obj.Name != "" {
+				name = obj.Name
+			}
+			filename := filepath.Join(artifactDir, fmt.Sprintf("%s.%s", name, "yaml"))
+			if obj.Path != "" {
+				subFolder := filepath.Join(artifactDir, obj.Path)
+				if err := os.MkdirAll(subFolder, 0755); err != nil && !os.IsExist(err) {
+					return fmt.Errorf("failed to create directory")
+				}
+				filename = filepath.Join(subFolder, fmt.Sprintf("%s.%s", name, "yaml"))
+			}
+			if err := pa.generateOutput(filename, obj.Object); err != nil {
 				return err
 			}
 		}
+		// if we have a local resource, write out the kustomization yaml limiting its visibility.
 		if artifact.Kustomize != nil {
-			data, err := yaml.Marshal(artifact.Kustomize)
-			if err != nil {
-				return err
-			}
-			filename := filepath.Join(artifactDir, "kustomization.yaml")
-			err = os.WriteFile(filename, data, 0644)
-			if err != nil {
+			// This is helmRelease related so it must be inside the sub-folder for the helm release.
+			filename := filepath.Join(artifactDir, artifact.SubFolder, "kustomization.yaml")
+			if err := writeOutKustomizeResource(artifact.Kustomize, filename); err != nil {
 				return err
 			}
 		}
+		filename := filepath.Join(artifactDir, "kustomization.yaml")
+		if err := writeOutKustomizeResource(artifact.KustomizeWrapper, filename); err != nil {
+			return err
+		}
 	}
 	return pa.generateOutput(filepath.Join(profileRootdir, "profile-installation.yaml"), &installation)
 }
 
+// writeOutKustomizeResource writes out kustomization resource data if set to a specific file.
+func writeOutKustomizeResource(kustomize *types.Kustomization, filename string) error {
+	data, err := yaml.Marshal(kustomize)
+	if err != nil {
+		return fmt.Errorf("failed to marshal kustomize resource: %w", err)
+	}
+	err = os.WriteFile(filename, data, 0644)
+	if err != nil {
+		return fmt.Errorf("failed to write file %s: %w", filename, err)
+	}
```

Comment:
ahem

![image](https://user-images.githubusercontent.com/8898786/126168717-4dd00bad-fe63-41be-bfdf-8f1d5f2dbdf2.png)

Reply:
To be fair, that function was an extract of code from something Jake wrote. :D :D :D :D

But superb catch right there in my face. :D

Reply:
WOW BLAME ME HUH :joy: :heart: 

Its your name attached to the git blame once this is merged :wink: